Wikipedia/Code/ArticleCollectionViewController.swift

```diff
@@ -101,6 +101,19 @@ class ArticleCollectionViewController: ColumnarCollectionViewController, Reading
     var eventLoggingLabel: EventLoggingLabel? {
         return nil
     }
+
+    var eventLoggingIndex: NSNumber? {
+        guard let index = previewedIndexPath?.item else {
+            return nil
+        }
+        return NSNumber(value: index)
+    }
+
+    var contentGroup: WMFContentGroup?
```

Comment:
If a property on this Core Data object is accessed after the object is deleted, it causes a crash. You could either observe core data changes and nil out this property if the content group is deleted (I don't think this makes sense in this case) OR just transfer the analytics properties you need as strings or some other non-core data class